compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3255,19 +3255,24 @@ TR::Register *OMR::X86::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, T
    TR::Register *srcPtrReg, *dstPtrReg, *transTableReg, *termCharReg, *lengthReg;
    bool stopUsingCopyReg1 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(0), srcPtrReg, cg);
    bool stopUsingCopyReg2 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(1), dstPtrReg, cg);
-   bool stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+   bool stopUsingCopyReg4 = false;
+   if (!sourceByte)
+      {
+      stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+      }
```

Comment:
Before your change child(3) was always evaluated.  Now its only evaluated conditionally if `sourceByte` is false.  If `sourceByte` is true its reference count is simply decremented later.  Is that what you intended?  If so, please explain.  There is a comment nearby that says if the source is a byte then child(3) is a "dummy".  I'm not sure what that means, but I'll assume that's why you skip the evaluation.  But please do clarify.